prog/dftb+/lib_dftb/energies.F90

```diff
@@ -33,6 +33,9 @@ module dftbp_energies
     !> spin energy
     real(dp) :: Espin   = 0.0_dp
 
+    !> range-separation energy
+    real(dp) :: Efock   = 0.0_dp
+
```

Comment:
Here, I only add the energy variable 'Efock' to express range-separation energy.
In this case, current code do not include any atom-resolved LC energy.
Is it possible to calculate atom-resolved this LC energy?

Reply:
For your algorithm this should be possible, but may need some care so that it agrees with other choices. 

For the evaluateLrEnergyDirect routine later in the code, this energy can be resolved as the coulombic energies of the exchange charges. Around line 1400 :

energy = energy + tmp * this%lrGammaEval(iAt1,iAt2)

replaced with

egyTmp = tmp * this%lrGammaEval(iAt1,iAt2)
energy = energy + egyTmp
Eatom(iAt1) = Eatom(iAt1) + 0.5_dp * egyTmp
Eatom(iAt2) = Eatom(iAt2) + 0.5_dp * egyTmp

I _think_ this is equivalent to a partial summation over the (Dmat * Hlr) that you use, but this would need to be tested for equivalence:

do iAt = 1, nAtom
  iStart = iSquare(iAt)
  iEnd = iSquare(iAt+1)-1
  Eatom(iAt1) = Eatom(iAt1) + sum(Dmat(:, iStart:iEnd) * Hlr(:, iStart:iEnd))
end do

Reply:
Just numerically checked this. There is a slight difference between the partial summation method and the exchange change energies (~1%), so they are not equivalent.

Reply:
In current step, it seems to be hard to compute atom-resolved LC energies in this algorithm. How about printing the warning sentence in stdout when AtomResolvedEnergies is 'Yes'?

Reply:
Yes, I think we should print a warning and switch off `AtomResolvedEnergies` in this case.

Reply:
Did you have a plan to implement to compute atom resolved LC energies for NeighbourBased or Thresholded cases? If not, then I will switch off AtomResolvedEnergies when we use LC calculation.

Reply:
I think switching it off would be the best solution for the moment. We can still re-enable it for LC once it had been implemented.

Reply:
Then current commit already did it. When we use LC with atom resolved energies, tAtomicResolved set to false. After implementing it, I will remove this part. Could you inform me when this PR is merged to master branch? As long as this PR is merged, I will start to merge ssr part.